sklearn/multiclass.py

```diff
@@ -980,10 +980,25 @@ def decision_function(self, X):
         predictions = np.vstack(
             [est.predict(Xi) for est, Xi in zip(self.estimators_, Xs)]
         ).T
-        confidences = np.vstack(
-            [_predict_binary(est, Xi) for est, Xi in zip(self.estimators_, Xs)]
-        ).T
-        Y = _ovr_decision_function(predictions, confidences, len(self.classes_))
+
+        confidences, probabilities_indexes = [], []
```

Comment:
I'm not a big fan of this since it really look like the code of `_predict_binary`. I'll try to open a first PR to use `_response_method_binary` instead of the `_predict_binary` because it does exactly the same job.

Once we have the refactor, I can see better what we need to improve here.

Reply:
This was the only way I saw to fix the bug, specifically to obtain the `probabilities_indexes`, but I understand your point. Looking forward to the refactor.